packages/ref-data-uploaders/cleanup-roadworks/index.ts

```diff
@@ -0,0 +1,57 @@
+import { Database, getDbClient, waitForDb } from "@reference-data-service/core/db";
+import { randomUUID } from "crypto";
+import * as logger from "lambda-log";
+import { Kysely, sql } from "kysely";
+
+export const deleteOldRoadworks = async (dbClient: Kysely<Database>) => {
+    await dbClient
+        .deleteFrom("roadworks")
+        .where(sql`DATE_ADD(DATE_FORMAT(left(actualEndDateTime,10),'%Y-%m-%d'), INTERVAL 7 DAY)`, "<", sql`CURDATE()`)
+        .where((query) =>
```

Comment:
This query isn't quite right, as it is it won't delete revoked or refused permit unless that have an actualEndDateTime (looks at the preprod DB the actualEndDateTime is usually null)

It's a bit complicated but the query should look like this:
```
 await dbClient
        .deleteFrom("roadworks")
        .where((qb) =>
            qb.or([
                qb.and([
                    qb(
                        sql`DATE_ADD(DATE_FORMAT(left(actualEndDateTime,10),'%Y-%m-%d'), INTERVAL 7 DAY)`,
                        "<",
                        sql`CURDATE()`,
                    ),
                    qb("permitStatus", "=", "closed"),
                ]),
                qb.and([
                    qb(
                        sql`DATE_ADD(DATE_FORMAT(left(lastUpdatedDateTime,10),'%Y-%m-%d'), INTERVAL 7 DAY)`,
                        "<",
                        sql`CURDATE()`,
                    ),
                    qb.or([
                        qb("permitStatus", "=", "revoked"),
                        qb("permitStatus", "=", "refused"),
                        qb("permitStatus", "=", "cancelled"),
                    ]),
                ]),
            ]),
        )
        .execute();
        ```
        
        I've also added the cancel permitStatus

Reply:
done